File: source_funcs/go_transcoder/LOWER_INSERTION_POINT/LOWER_INSERTION_POINT.go
package main

func f_gold(arr []int, n int, X int) int {
	if X < arr[0] {
		return 0
	} else if X > arr[n-1] {
		return n
	}
	var lowerPnt int = 0
	var i int = 1
	for i < n && arr[i] < X {
		lowerPnt = i
		i = i * 2
	}
	for lowerPnt < n && arr[lowerPnt] < X {
		lowerPnt++
	}
	return lowerPnt
}

File: rust_funcs/c_transcoder/LOWER_INSERTION_POINT/LOWER_INSERTION_POINT.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(arr:[f32;2], n:u32, x:f32) -> i32 {
    if x < arr[0] {
        return 0; 
    } else if x > arr[n-1]  {
        return n;
    }
    let mut lower_pnt = 0;
    let mut i = 1;
    while i < n && arr[i] < x {
        lower_pnt = i;
        i *= 2;
    }
    while lower_pnt < n && arr[lower_pnt] < x {
        lower_pnt += 1;
    }
    lower_pnt
}
